pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -353,8 +394,31 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(
 			// TODO: this should be parsed from the defaultEndpoint field in the ingressListener
 			instance.Endpoint.Port = listenPort.Port
 
+			// Protocol sniffing for inbound listener.
+			// For each ingress listener, find a service instance with the same port number and override the service port.
+			// The listener port protocol is determined by the service port protocol. If user doesn't specify the service
+			// port protocol, the listener will be generated using protocol sniffing.
+			// For example, the set of service instances
+			//      --> NetworkEndpoint
+			//              Address:Port 172.16.0.1:1111
+			//              ServicePort  80|HTTP
+			//      --> NetworkEndpoint
+			//              Address:Port 172.16.0.1:2222
+			//	            ServicePort  8888|TCP
+			//      --> NetworkEndpoint
+			//              Address:Port 172.16.0.1:3333
+			//              ServicePort 9999|Unknown
+			//
+			// User defines ingress listener as:
+			//  ingress:
+			//	- port:
+			//      number: 1111
+			//      protocol: unknown
+			//    defaultEndpoint: unix://somedir
+			// Only one listener will be generated. The the listener port protocol is unknown.
+			//
```

Comment:
remove this comment (in this or another PR). This comment is not applicable here as validation ensures that the protocol specified in Sidecar.ingress is always a valid known protocol.

Reply:
Removed